common/rawaccel.hpp

```diff
@@ -46,27 +45,228 @@ namespace rawaccel {
             clamp_speed = args.speed_max > 0 && args.speed_min <= args.speed_max;
             apply_rotate = args.degrees_rotation != 0;
             apply_snap = args.degrees_snap != 0;
-            apply_directional_weight = args.whole && 
+            apply_directional_weight = args.speed_processor_args.whole && 
                 args.range_weights.x != args.range_weights.y;
             compute_ref_angle = apply_snap || apply_directional_weight;
             apply_dir_mul_x = args.lr_sens_ratio != 1;
             apply_dir_mul_y = args.ud_sens_ratio != 1;
+        }
+
+        modifier_flags() = default;
+    };
+
+    /// <summary>
+    /// Simple exponential moving average smoother.
+    /// </summary>
+    struct simple_ema_smoother {
+
+        double windowCoefficient = 0;
+        double cutoffCoefficient = 0;
+
+        void init(const double halfLife)
+        {
+            windowTotal = 0;
+            cutoffTotal = 0;
+
+            windowCoefficient = halfLife > 0 ? pow(0.5, 1 / halfLife) : 0;
+            cutoffCoefficient = 1.0 - sqrt(1.0 - windowCoefficient);
+        }
+
+        double windowTotal = 0;
+        double cutoffTotal = 0;
+
+        double smooth(const double speed, const milliseconds time)
+        {
+            // compute coefficients
+            double timeAdjustedWindowCoefficient = 1 - pow(windowCoefficient, time);
+            double timeAdjustedCutoffCoefficient = 1 - pow(cutoffCoefficient, time);
+
+            // adjust total based on coefficient and difference between new value and total
+            windowTotal += timeAdjustedWindowCoefficient * (speed - windowTotal);
+            cutoffTotal += timeAdjustedWindowCoefficient * (speed - cutoffTotal);
+
+            return min(windowTotal, cutoffTotal);
+        }
+    };
+
+    /// <summary>
+    /// Linear exponential moving average smoother.
+    /// </summary>
+    struct linear_ema_smoother {
+
+        // This constant found via experimentation.
+        // Allowing user to specify may confuse parameterization without much gain.
+        const double trendDampening = 0.75;
```

Comment:
use `static constexpr` for static members
```suggestion
        static constexpr double trendDampening = 0.75;
```